NuRadioReco/modules/io/eventWriter.py

```diff
@@ -126,26 +136,40 @@ def __get_event_bytearray(self, event, mode):
         return event_bytearray
 
     def __get_detector_dict(self, event, det):
+        is_generic_detector = isinstance(det, generic_detector.GenericDetector)
         det_dict = {
+            "generic_detector": is_generic_detector,
             "channels": {},
             "stations": {}
         }
+        if is_generic_detector:
+            det_dict['default_station'] = det.get_default_station_id()
+            det_dict['default_channel'] = det.get_default_channel_id()
         i_station = 0
         i_channel = 0
         for station in event.get_stations():
-            if not self.__is_station_already_in_file(station):
-                det.update(station.get_station_time())
-                station_description = det.get_station(station.get_id())
-                det_dict['stations'][str(i_station)] = station_description
-                self.__stored_stations.append({
+            if not self.__is_station_already_in_file(station.get_id(), station.get_station_time()):
+                if not is_generic_detector:
+                    det.update(station.get_station_time())
+                    station_description = det.get_station(station.get_id())
+                    self.__stored_stations.append({
                     'station_id': station.get_id(),
                     'commission_time': station_description['commission_time'],
                     'decommission_time': station_description['decommission_time']
-                })
+                    })
+                else:
+                    station_description = det.get_raw_station(station.get_id())
```

Comment:
I don't understand what a "raw_station" is. Please add an explanation to the doc string in generic detector.